packages/typescript/src/core/trace_manager.ts

```diff
@@ -0,0 +1,144 @@
+import { LiveSpan, Span } from './entities/span';
+import { TraceInfo } from './entities/trace_info';
+import { Trace } from './entities/trace';
+import { TraceData } from './entities/trace_data';
+import { SpanAttributeKey } from './constants';
+
+/**
+ * Internal representation to keep the state of a trace.
+ * Uses a Map<string, LiveSpan> instead of TraceData to allow access by span_id.
+ */
+class _Trace {
+  info: TraceInfo;
+  spanDict: Map<string, LiveSpan>;
+
+  constructor(info: TraceInfo) {
+    this.info = info;
+    this.spanDict = new Map<string, LiveSpan>();
+  }
+
+  /**
+   * Convert the internal trace representation to an MLflow Trace object
+   */
+  toMlflowTrace(): Trace {
+    const traceData = new TraceData();
+
+    // Convert LiveSpan, mutable objects, into immutable Span objects before persisting
+    for (const span of this.spanDict.values()) {
+      traceData.spans.push(span as Span);
+    }
```

Comment:
nit: can we do this?
```suggestion
    const traceData = new TraceData([...this.spanDict.values()] as Span[]);
```